### DISCUSSION TITLE
Allow Order Discount Function to combine with other coupons via parameters

### DISCUSSION THREAD

**Discussion by EtienneGELAS** [Shopify/function-examples#363] [Feedback: Discounts APIs]
**Summary**

Shopify's stacking discount codes feature lacks compatibility with apps that auto-generate discounts, such as Klaviyo, making bulk coupon combinations challenging.

**Details**

- Shopify's Discount System Evolution: Previously independent Shopify scripts have now been incorporated into the discount panel as Shopify functions auto discount.
- Discount Stacking Option: Each discount code setting now offers an option to dictate if the code can be combined with other individual or auto-generated codes.
- Lack of Functionality in Auto-Generate Apps: Predominantly, apps like Klaviyo that auto-generate discount codes don't provide a setting to enable or disable stacking for these codes.
- Manual Overhead for Merchants: The absence of this feature compels merchants to manually adjust possibly vast numbers of previously created codes to accommodate the new stacking specifications.
- Potential for Unchecked Discounts: If individual codes can freely combine with automatic discounts, this can potentially lead to unchecked and hefty discounts. Imagine a scenario where an email discount is used in conjunction with an influencer discount and an auto quantitative discount.

**Illustrative Use Case**

Consider a Shopify Plus merchant with a cart-level progress bar detailing different spending thresholds that grant specific percentage discounts. Earlier, these were facilitated using scripts that operated externally to the primary discount engine. This meant, upon checkout, any coupon could be combined since script discounts weren't treated as discount codes. Consequently, a customer spending a certain amount could avail of a specific percentage discount in addition to another coupon code (e.g., an influencer code).

With the transition to functions, all such discounts are now processed via the primary discount engine. If a merchant implements a function-based discount (e.g., X% off for spending Y$), this won't stack with other codes by default. This forces the merchant to manually modify all existing active coupons to sync with the function-based discount. Furthermore, auto-generated coupons (like those from Klaviyo) presently don't support combinable discount creation by default. This imposes a continuous manual adjustment burden on the merchant every time a new batch of codes is generated.

The most pressing issue arises when merchants allow all app-generated coupons (or other preexisting coupons) to stack with a function-based order discount. This inadvertently makes all individual coupons stackable. This can potentially allow a buyer to combine an auto discount from functions, a Klaviyo coupon, and an unlimited number of other coupons (like influencer codes).

**Suggested Solution:**

A potential solution to this issue would be to introduce a parameter or setting in Shopify functions that allows auto-generated discounts to be stacked with pre-existing store coupons, but prevents individual pre-existing coupons from stacking with each other.

**Comment by Damjan9898** [comment]
I found a solution for this problem. You can try this Shopify custom app: https://apps.shopify.com/syncdisc
